agent/agent.go

```diff
@@ -247,8 +253,6 @@ type Agent struct {
 	eventLock   sync.RWMutex
 	eventNotify NotifyGroup
 
-	reloadCh chan chan error
```

Comment:
The reloadCh was previously used for the odd interaction between the cli command and the Agent itself. To do an API triggered reload before required the agent endpoint code pushing a chan into this chan and then waiting to hear back that the reload was completed. The cli command was responsible for processing that chan.

Now reloads happen a little differently since the Agent itself is responsible for parsing configuration (or rather the autoconf package). Now the cli command will process a SIGHUP and just call ReloadConfig on the Agent. The HTTP endpoint will do similarly and there is no need for the channel.